File: fix_opacity.dart
import 'dart:io';

void main() {
  final dir = Directory('lib');
  if (!dir.existsSync()) {
    print('lib directory not found.');
    return;
  }

  int count = 0;
  for (final entity in dir.listSync(recursive: true)) {
    if (entity is File && entity.path.endsWith('.dart')) {
      final content = entity.readAsStringSync();
      final newContent = content.replaceAllMapped(
        RegExp(r'\.withOpacity\(\s*([^)]+)\s*\)'),
        (match) => '.withValues(alpha: ${match.group(1)})'
      );
      if (newContent != content) {
        entity.writeAsStringSync(newContent);
        print('Updated ${entity.path}');
        count++;
      }
    }
  }
  print('Updated $count files.');
}

File: fix_opacity.py
import os
import re

dir_path = r"d:\Development\projects\shopons\lib"

for root, dirs, files in os.walk(dir_path):
    for file in files:
        if file.endswith(".dart"):
            file_path = os.path.join(root, file)
            with open(file_path, "r", encoding="utf-8") as f:
                content = f.read()

            new_content = re.sub(r'\.withOpacity\(\s*([^)]+)\s*\)', r'.withValues(alpha: \1)', content)

            if new_content != content:
                with open(file_path, "w", encoding="utf-8") as f:
                    f.write(new_content)
                print(f"Updated {file_path}")
